xclim/indices/_agro.py

```diff
@@ -0,0 +1,101 @@
+# noqa: D100
+from typing import Optional
+
+import numpy as np
+import xarray
+
+from xclim.core.calendar import resample_doy
+from xclim.core.units import (
+    convert_units_to,
+    declare_units,
+    pint2cfunits,
+    rate2amount,
+    str2pint,
+    to_agg_units,
+)
+
+from . import run_length as rl
+from ._conversion import rain_approximation, snowfall_approximation
+from .generic import select_resample_op, threshold_count
+
+# Frequencies : YS: year start, QS-DEC: seasons starting in december, MS: month start
+# See http://pandas.pydata.org/pandas-docs/stable/timeseries.html#offset-aliases
+
+# -------------------------------------------------- #
+# ATTENTION: ASSUME ALL INDICES WRONG UNTIL TESTED ! #
+# -------------------------------------------------- #
+
+__all__ = ["corn_heat_units"]
+
+
+@declare_units(
+    tasmin="[temperature]",
+    tasmax="[temperature]",
+    thresh_tasmin="[temperature]",
+    thresh_tasmax="[temperature]",
+)
+def corn_heat_units(
+    tasmin: xarray.DataArray,
+    tasmax: xarray.DataArray,
+    thresh_tasmin: str = "4.44 degC",
+    thresh_tasmax: str = "10 degC",
+) -> xarray.DataArray:
+    r"""Corn heat units.
+
+    Temperature-based index used to estimate the development of corn crops.
+
+    Parameters
+    ----------
+    tasmin : xarray.DataArray
+      Minimum daily temperature.
+    tasmax : xarray.DataArray
+      Maximum daily temperature.
+    thresh_tasmin : str
+      The minimum temperature threshold needed for corn growth.
+    thresh_tasmax : str
+      The maximum temperature threshold needed for corn growth.
+
+    Returns
+    -------
+    xarray.DataArray, [dimensionless]
+      Daily corn heat units.
+
+    Notes
+    -----
+    The thresholds of 4.44°C for minimum temperatures and 10°C for maximum temperatures were selected following
+    the assumption that no growth occurs below these values.
+
+    Let :math:`TX_{i}` and :math:`TN_{i}` be the daily maximum and minimum temperature at day :math:`i`. Then the daily
+    corn heat unit is:
+
+    .. math::
+        CHU_i = \frac{YX_{i} + YN_{i}}{2}
+
+    with
+
+    .. math::
+
+        YX_i & = 3.33(TX_i -10) - 0.084(TX_i -10)^2, &\text{if } TX_i > 10°C
+
+        YN_i & = 1.8(TN_i -4.44), &\text{if } TN_i > 4.44°C
+
+    where :math:`YX_{i}` and :math:`YN_{i}` is 0 when :math:`TX_i \leq 10°C` and :math:`TN_i \leq 4.44°C`, respectively.
+    """
+
+    thresh_tasmin = convert_units_to(thresh_tasmin, tasmin)
+    thresh_tasmax = convert_units_to(thresh_tasmax, tasmax)
```

Comment:
Héhé, what if tasmin is in °F and tasmax in °C ? 

En fait, doesn't that algorithm _expect_ Celsius? I understand K are ok because of the subtraction, but would suggest hard-converting everything to cover the corner cases. Here, there is no need to be flexible as for output units, since they are inherent to the indice.

Reply:
You are right, the algorithm expect Celsius, I added conversion of all temperature input in Celsius.